packages/api-server/src/handler.ts

```diff
@@ -1,31 +1,129 @@
-import { http } from './http'
-import type { HttpServerParams } from './http'
+import { getConfig } from '@redwoodjs/internal'
 
-export const cliOptions = {
+import createApp from './app'
+import withApiProxy from './middleware/withApiProxy'
+import withFunctions from './middleware/withFunctions'
+import withWebServer from './middleware/withWebServer'
+import { startServer } from './server'
+import type { HttpServerParams } from './server'
+
+/*
+ * This file has defines CLI handlers used by the redwood cli, for `rw serve`
+ * Also used in index.ts for the api server
+ */
+
+export const commonOptions = {
+  port: { default: 8910, type: 'number', alias: 'p' },
+  socket: { type: 'string' },
+} as const
+
+export const apiCliOptions = {
   port: { default: 8911, type: 'number', alias: 'p' },
   socket: { type: 'string' },
-  rootPath: {
-    alias: 'root-path',
+  apiRootPath: {
+    alias: ['rootPath', 'root-path'],
     default: '/',
     type: 'string',
     desc: 'Root path where your api functions are served',
-    coerce: (path: string) => {
-      // Make sure that we create a root path that starts and ends with a slash (/)
-      const prefix = path.charAt(0) !== '/' ? '/' : ''
-      const suffix = path.charAt(path.length - 1) !== '/' ? '/' : ''
+    coerce: coerceRootPath,
+  },
+} as const
 
-      return `${prefix}${path}${suffix}`
-    },
+export const webCliOptions = {
+  port: { default: 8910, type: 'number', alias: 'p' },
+  socket: { type: 'string' },
+  apiHost: {
+    alias: 'api-host',
+    type: 'string',
+    desc:
+      'Forward requests from the apiProxyPath, defined in redwood.toml to this host',
   },
 } as const
 
-export const handler = ({ port, socket, rootPath }: HttpServerParams) => {
-  http({ port, socket, rootPath }).on('listening', () => {
+interface ApiServerArgs extends Omit<HttpServerParams, 'app'> {
+  apiRootPath: string // either user supplied or '/'
+}
+
+export const apiServerHandler = ({
+  port,
+  socket,
+  apiRootPath,
+}: ApiServerArgs) => {
+  let app = createApp()
+
+  // Attach middleware
+  app = withFunctions(app, apiRootPath)
+
+  startServer({
+    port,
+    socket,
+    app,
+  }).on('listening', () => {
     if (socket) {
       console.log(`Listening on ${socket}`)
-    } else {
-      console.log(`Listening on http://localhost:${port}${rootPath}`)
     }
-    console.log()
+    console.log(`APIs Listening on http://localhost:${port}${apiRootPath}`)
```

Comment:
I use this during development, and I already append `api` in-front of the log output.
```suggestion
    console.log(`Listening on http://localhost:${port}${apiRootPath}`)
```